src/couch_index.erl

```diff
@@ -407,3 +394,160 @@ assert_signature_match(Mod, OldIdxState, NewIdxState) ->
         {Sig, Sig} -> ok;
         _ -> erlang:error(signature_mismatch)
     end.
+
+commit_compacted(NewIdxState, State) ->
+    #st{
+        mod=Mod,
+        idx_state=OldIdxState,
+        updater=Updater,
+        commit_delay=Delay
+    } = State,
+    {ok, NewIdxState1} = Mod:swap_compacted(OldIdxState, NewIdxState),
+    % Restart the indexer if it's running.
+    case couch_index_updater:is_running(Updater) of
+        true -> ok = couch_index_updater:restart(Updater, NewIdxState1);
+        false -> ok
+    end,
+    case State#st.committed of
+        true -> erlang:send_after(Delay, self(), commit);
+        false -> ok
+    end,
+    State#st{
+        idx_state=NewIdxState1,
+        committed=false
+     }.
+
+is_recompaction_enabled(IdxState, #st{mod = Mod}) ->
+    DbName = binary_to_list(Mod:get(db_name, IdxState)),
+    IdxName = binary_to_list(Mod:get(idx_name, IdxState)),
+    IdxKey = DbName ++ ":" ++ IdxName,
+    Global = config:get("view_compaction.recompaction", "enable"),
+    PerIndex = config:get("view_compaction.recompaction", IdxKey),
+    PerDb = config:get("view_compaction.recompaction", DbName),
+    case {Global, PerDb, PerIndex} of
```

Comment:
This truth table is confusing to me. I'd disable to on and then have a trigger off with a simple orelse combo. Something along the lines of:

    Global = config:get("view_compaction.recompaction", "enabled", "true") == "false",
    PerDb = config:get("view_compaction.recompaction", DbName, "enabled") == "disabled",
    PerIdx = config:get("view_compaction.recompaction", IdxKey, "enabled") == "disabled",
    Global orelse PerDb orelse PerIdx.